app/services/users/omniauth.rb

```diff
@@ -0,0 +1,37 @@
+class Users::Omniauth
+  attr_reader :auth
+
+  def initialize(auth)
+    @auth = auth
+  end
+
+  def get
+    user
+  end
+
+  private
+
+  delegate :provider, :uid, :info, to: :auth
+  delegate :email, to: :info
+
+  def user
+    user = find_user
+    user ||= create_user
+    user
+  end
+
+  def find_user
+    User.where(provider: provider, uid: uid, email: email).first
+  end
+
+  def create_user
+    user = User.new(
+      provider: provider,
+      uid: uid,
+      email: email,
+      password: Devise.friendly_token[0, 20]
+    )
+    user.skip_confirmation!
+    user.save
+  end
+end
```

Comment:
@sinpai let's add a unit test for `Users::Omniauth`. The test file can be located in `rspec/unit/services/users/omniauth_test.rb`